File: module-assembler-common/com/electrolytej/assembler/model/Module.java
package com.electrolytej.assembler.model;


/**
 * simpleName : app
 * canonicalName : :app
 * format : app
 * group : father
 * binary_artifact : com.jamesfchen.b:app:1.0
 * deps : [":hotel-module:bundle1",":hotel-module:bundle2"]
 */
public class Module {
    public String group;
    public String simpleName;
    public String sourcePath;
    public String binaryPath;
    public String projectDir;
    public String format;
    public String versionName;
    public String versionCode;
    public String dynamic;
    public Type type = Type.NONE;

    //    public String binary_artifact;
    public enum Type {
        NONE,

        //    AAR,
        //    JAR,
        BINARY, SOURCE, EXCLUDE
    }
}

File: module-assembler-gradle-plugin-kt/src/main/kotlin/com/electrolytej/assembler/model/Module.kt
package com.electrolytej.assembler.model

import kotlinx.serialization.Serializable

/**
 * simpleName : app
 * canonicalName : :app
 * format : app
 * group : father
 * binary_artifact : com.jamesfchen.b:app:1.0
 * deps : [":hotel-module:bundle1",":hotel-module:bundle2"]
 */
@Serializable
data class Module(
    var group: String,
    var simpleName: String,
) {
    var sourcePath: String = ""
    var binaryPath: String = ""
    var projectDir: String = ""
    var format: String = ""
    var versionName: String = ""
    var versionCode: String = ""
    var dynamic: String? = null
    var type: Type = Type.NONE

    //    public String binary_artifact;


    enum class Type {
        NONE,

        //    AAR,
        //    JAR,
        BINARY,
        SOURCE,
        EXCLUDE
    }
}